packages/core/src/app/order/OrderSummaryItems.tsx

```diff
@@ -23,115 +23,87 @@ export interface OrderSummaryItemsProps {
-                                <OrderSummaryItem {...summaryItemProps} />
-                            </li>
-                        ))}
-                </ul>
-
-                {this.renderActions()}
-            </>
-        );
-    }
-
-    private getCollapsedLimit(): number {
-        return isSmallScreen() ? COLLAPSED_ITEMS_LIMIT_SMALL_SCREEN : COLLAPSED_ITEMS_LIMIT;
-    }
-
-    private renderActions(): ReactNode {
-        const { isExpanded } = this.state;
-
-        if (this.getLineItemCount() <= this.getCollapsedLimit()) {
-            return;
-        }
-
-        return (
-            <div className="cart-actions">
-                <button
-                    className={classNames('button button--tertiary button--tiny optimizedCheckout-buttonSecondary',
-                        { 'sub-text-medium': this.props.themeV2 })}
-                    onClick={this.handleToggle}
-                    type="button"
-                >
-                    {isExpanded ? (
-                        <>
-                            <TranslatedString id="cart.see_less_action" />
-                            <IconChevronUp />
-                        </>
-                    ) : (
-                        <>
-                            <TranslatedString id="cart.see_all_action" />
-                            <IconChevronDown />
-                        </>
-                    )}
-                </button>
-            </div>
-        );
-    }
-
-    private getLineItemCount(): number {
-        const { items } = this.props;
-
-        return (
-            (items.customItems || []).length +
-            items.physicalItems.length +
-            items.digitalItems.length +
-            items.giftCertificates.length
-        );
-    }
-
-    private handleToggle: () => void = () => {
-        const { isExpanded } = this.state;
-
-        this.setState({ isExpanded: !isExpanded });
-    };
-}
+                </h3>
+            )}
+
+            <ul aria-live="polite" className="productList">
+                {[
+                    ...items.physicalItems
+                        .slice()
+                        .sort((item) => item.variantId)
+                        .map(mapFromPhysical),
+                    ...items.giftCertificates.slice().map(mapFromGiftCertificate),
+                    ...items.digitalItems
+                        .slice()
+                        .sort((item) => item.variantId)
+                        .map(mapFromDigital),
+                    ...(items.customItems || []).map(mapFromCustom),
+                ]
+                    .slice(0, isExpanded ? undefined : collapsedLimit)
+                    .map((summaryItemProps) => (
+                        <li className="productList-item is-visible" key={summaryItemProps.id}>
+                            <OrderSummaryItem {...summaryItemProps} />
+                        </li>
+                    ))}
+            </ul>
+
+            {shouldShowActions && (
```

Comment:
This also can we refactored to another component.

Reply:
I have created three small components to enhance code readability.